literals/literals.go

```diff
@@ -0,0 +1,240 @@
+package literals
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	mathrand "math/rand"
+	"strconv"
+
+	"golang.org/x/tools/go/ast/astutil"
+	"mvdan.cc/garble/literals/obfuscators"
+)
+
+func callExpr(resultType ast.Expr, block *ast.BlockStmt) *ast.CallExpr {
+	return &ast.CallExpr{
+		Fun: &ast.FuncLit{
+			Type: &ast.FuncType{
+				Params: &ast.FieldList{},
+				Results: &ast.FieldList{
+					List: []*ast.Field{
+						{Type: resultType},
+					},
+				},
+			},
+			Body: block,
+		},
+	}
+}
+
+func randObfuscator() obfuscators.Obfuscator {
+	randPos := mathrand.Intn(len(obfuscators.Obfuscators))
+	return obfuscators.Obfuscators[randPos]
+}
+
+func returnStmt(result ast.Expr) *ast.ReturnStmt {
+	return &ast.ReturnStmt{
+		Results: []ast.Expr{result},
+	}
+}
+
+func isTypeDefStr(typ types.Type) bool {
+	strType := types.Typ[types.String]
+
+	if named, ok := typ.(*types.Named); ok {
+		return types.Identical(named.Underlying(), strType)
+	}
+
+	return false
+}
+
+func containsTypeDefStr(expr ast.Expr, info *types.Info) bool {
```

Comment:
please document this; why do we care if an expression contains a string type?

also, I think you should replace `TypeDefStr` with something like `StringUnderlying`, since "underlying" is the correct terminology for what you're doing here

Reply:
We don't need it we can just ask go/types on the literal expresion